src/api/model/modelApi.ts

```diff
@@ -14,57 +14,91 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
+import _ from 'lodash';
 
-export class ModelApi extends TransactionApi {
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
   async installModels(database: string, engine: string, models: Model[]) {
-    const action: InstallAction = {
-      type: 'InstallAction',
-      sources: models,
-    };
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """ ${model.value} """`;
+    });
 
-    return await this.runActions(database, engine, [action], false);
+    return await this.exec(database, engine, queries.join('\n'), [], false);
+  }
+
+  async installModelsAsync(database: string, engine: string, models: Model[]) {
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """ ${model.value} """`;
+    });
+
+    return this.execAsync(database, engine, queries.join('\n'), [], false);
   }
 
   async listModels(database: string, engine: string) {
-    const action: ListSourceAction = {
-      type: 'ListSourceAction',
-    };
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const outName = `models${randInt}`;
+    const rsp = await this.exec(
+      database,
+      engine,
+      `def output:${outName}[name] = rel:catalog:model(name, _)`,
+    );
+
+    const models = rsp.results.map(result => {
+      if (result.relationId.includes('/:output/:models')) {
+        return result.table.toArray().map(col => {
+          return col.v1;
+        });
+      }
+    })[0];
 
-    return await this.runActions(database, engine, [action]);
+    return models;
   }
 
   async getModel(database: string, engine: string, name: string) {
-    const result = await this.listModels(database, engine);
+    const randInt = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+    const outName = `model${randInt}`;
+    const rsp = await this.exec(
+      database,
+      engine,
+      `def output:${outName} = rel:catalog:model["${name}"]`,
+    );
 
-    if (result.actions[0]?.result?.type === 'ListSourceActionResult') {
-      const model = result.actions[0].result.sources.find(m => m.name === name);
-
-      if (model) {
-        return model;
+    const value = rsp.results.map(result => {
+      if (result.relationId.includes(`/:output/:${outName}`)) {
+        return result.table.toArray().map(col => {
+          return col.v1;
+        });
       }
+    });
+
+    if (value.length == 0 || value[0] == undefined) {
+      throw new Error(`Model '${name}' not found`);
     }
 
-    throw new Error(`Model '${name}' not found`);
+    return { name: name, value: value[0][0] } as Model;
```

Comment:
Type casting here breaks type safety here. 

```
export type Model = {
  name: string;
  value: string;
  type: string;
  path?: string;
};
```

`type` is a required property.

But, in v2 it should be 

```
export type Model = {
  name: string;
  value: string;
};
```

so then you can re-write the code like

```
const model: Model = {name: name, value: value[0][0]}

return model;
```